### PR TITLE
BSM geospatial query

### PR DESCRIPTION
**Pull request by Michael7371**
Changes in this PR:

- Changed the get "/bsm/json" endpoint to include a geospatial query to limit the number of bsms being returned to the frontend.
- Fixed unit tests for Map, Spat, and the bsm endpoint
- Added changes to the frontend to update the request arguments to include a long, lat, and query distance.

### CODE REVIEW

File: gui/src/apis/mm-api.ts [hunk 1]
```diff
@@ -68,18 +68,27 @@ class MessageMonitorApi {
     vehicleId,
     startTime,
     endTime,
+    long,
```
Comment (jacob6838) [inline gui/src/apis/mm-api.ts:71]: Is this just a circular distance-based query? Why is it not sufficient to just query by incident ID?

Comment (jacob6838) [reply gui/src/apis/mm-api.ts:71]: By incident ID I mean intersection ID lol. If there isn't a link to an intersection, then this looks great

Comment (Michael7371) [reply gui/src/apis/mm-api.ts:71]: Yeah I was talking to John about using the originIp field but apparently one of the deployments has issues with the IP's being off for the intersections. So I decided to just do an offset lat/long based query to only get bsms in the general area of the intersection.

Comment (jacob6838) [reply gui/src/apis/mm-api.ts:71]: makes sense to me

**Review by jacob6838 — APPROVED** [review]

**Pull request merged**